plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -72,7 +72,15 @@
     private String _manageSnapshotPath;
 
     private String rbdTemplateSnapName = "cloudstack-base-snap";
-    private int rbdFeatures = (1 << 0); /* Feature 1<<0 means layering in RBD format 2 */
+    /*
+        Ceph RBD features to use when creating a new RBD image
+        (1 << 0): RBD_FEATURE_LAYERING
+        (1 << 2): RBD_FEATURE_EXCLUSIVE_LOCK
+        (1 << 3): RBD_FEATURE_OBJECT_MAP
+        (1 << 4): RBD_FEATURE_FAST_DIFF
+        (1 << 5): RBD_FEATURE_DEEP_FLATTEN
+     */
+    private int rbdFeatures = (1 << 0) + (1 << 2) + (1 << 3) + (1 << 4) + (1 << 5);
```

Comment:
How about define the feature as various final int hex values and `rbdFeatures` can be the or-ed value of those features. Should there also be some kind of version vs feature check?

Reply:
Yeah, I could also just write **61** as that's the default.

I took these values from the C-header file from Ceph so that people see what we actually use.

That's mainly a code style thing.

Reply:
Agree that it is mainly a code style thing. But here are some suggestions as well.
```
private final int RBD_FEATURE_LAYERING = 1;
private final int RBD_FEATURE_EXCLUSIVE_LOCK = 4;
private final int RBD_FEATURE_OBJECT_MAP = 8;
private final int RBD_FEATURE_FAST_DIFF = 16;
private final int RBD_FEATURE_DEEP_FLATTEN = 32;
private int rbdFeatures = RBD_FEATURE_LAYERING + RBD_FEATURE_EXCLUSIVE_LOCK + RBD_FEATURE_OBJECT_MAP + RBD_FEATURE_FAST_DIFF + RBD_FEATURE_DEEP_FLATTEN;
```
or
```
/**
	Ceph RBD features to use when creating a new RBD image
        RBD_FEATURE_LAYERING: 1
        RBD_FEATURE_EXCLUSIVE_LOCK: 4
        RBD_FEATURE_OBJECT_MAP: 8
        RBD_FEATURE_FAST_DIFF: 16
        RBD_FEATURE_DEEP_FLATTEN: 32
	Total:61
*/
private final int RBF_FEATURES = 61;
```

Reply:
I like the first suggestion @GabrielBrascher makes, @wido . it comes down to "self documenting code by proper naming"